### ISSUE TITLE
numpy.core.multiarray failed to import

### ISSUE THREAD

**Issue by HUAN5235** [numpy/numpy#14632]
I am trying to download opencv on my raspberry pi using pip3, but when i imported it it returned an error
```
>>> import cv2
ImportError: numpy.core.multiarray failed to import
Traceback (most recent call last):
  File "<pyshell#4>", line 1, in <module>
    import cv2
  File "/home/pi/.local/lib/python3.5/site-packages/cv2/__init__.py", line 3, in <module>
    from .cv2 import *
ImportError: numpy.core.multiarray failed to import
```

I googled some answers, like this [stack overflow answer](https://stackoverflow.com/questions/20518632/importerror-numpy-core-multiarray-failed-to-import) here. It told me to update numpy to the newest version and install some packages. I treid to import numpy again after updating but I met this error

```
>>> import numpy
Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.5/site-packages/numpy/core/__init__.py", line 17, in <module>
    from . import multiarray
  File "/home/pi/.local/lib/python3.5/site-packages/numpy/core/multiarray.py", line 14, in <module>
    from . import overrides
  File "/home/pi/.local/lib/python3.5/site-packages/numpy/core/overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ImportError: /home/pi/.local/lib/python3.5/site-packages/numpy/core/_multiarray_umath.cpython-35m-arm-linux-gnueabihf.so: undefined symbol: cblas_sgemm

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<pyshell#5>", line 1, in <module>
    import numpy
  File "/home/pi/.local/lib/python3.5/site-packages/numpy/__init__.py", line 142, in <module>
    from . import core
  File "/home/pi/.local/lib/python3.5/site-packages/numpy/core/__init__.py", line 47, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy c-extensions failed.
- Try uninstalling and reinstalling numpy.
- If you have already done that, then:
  1. Check that you expected to use Python3.5 from "/usr/bin/python3",
     and that you have no directories in your PATH or PYTHONPATH that can
     interfere with the Python and numpy version "1.17.2" you're trying to use.
  2. If (1) looks fine, you can open a new issue at
     https://github.com/numpy/numpy/issues.  Please include details on:
     - how you installed Python
     - how you installed numpy
     - your operating system
     - whether or not you have multiple versions of Python installed
     - if you built from source, your compiler versions and ideally a build log

- If you're working with a numpy git repository, try `git clean -xdf`
  (removes all files not under version control) and rebuild numpy.

Note: this error has many possible causes, so please don't comment on
an existing issue about this - open a new one instead.

Original error was: /home/pi/.local/lib/python3.5/site-packages/numpy/core/_multiarray_umath.cpython-35m-arm-linux-gnueabihf.so: undefined symbol: cblas_sgemm
```
I have python 2, 3.5 and 3.6 on my raspi, so I uninstalled numpy on all versions of python and reinstalled it on python 3.5 only, but it returned the same error. I am using raspbian os and installed numpy using pip3. How can I solve this problem?

**Comment by HUAN5235** [comment]
It turns out that I haven't uninstalled numpy on python 2, but after i did it i got the same error.

**Comment by mattip** [comment]
We do not create the pip-installable wheels. You are probably downloading them from [piwheels](https://www.piwheels.org/). Did you follow the instructions on [their blog](https://blog.piwheels.org/new-opencv-builds/) ?

**Comment by HUAN5235** [comment]
no, but ill try. thanks

**Comment by HUAN5235** [comment]
It spat out the same error. What should I do now? I thought i saw the command in the piwheels blog try to download an older version of numpy, so I tried uninstalling all numpy files and reinstalling again. But then I met another error saying permission denied by some file called f2py at /usr/local/bin when trying to uninstall numpy. There are also f2py3 and f2py3.5 files in the folder, are they somehow connected? Should I try to reinstall numpy, or use other ways to download opencv? I have found a tutorial on manually installing opencv then building it [here](https://docs.opencv.org/3.4/d2/de6/tutorial_py_setup_in_ubuntu.html). Should I follow it?

**Comment by mattip** [comment]
Building OpenCV and NumPy on a raspberry pi is very painful. I would ask on the forums connected with piwheels or raspberry pi, it seems one of your support library installations is out of date or corrupted. You should not be using `pip` to install and uninstall things in `/usr/local/bin`, that is the province of `apt`. `pip` should be installing to a virtualenv or to`~/.local/lib/pythonx.x/site-packages`

**Comment by kzy77** [comment]
I also encountered this problem, my system is mac, I installed python 2.7 and python 3.7.5. I guess it's a multi-environment problem.

**Comment by kzy77** [comment]
But I import as well in pycharm

**Comment by mattip** [comment]
@HUAN5235 did you solve your problems installing NumPy?

**Comment by HUAN5235** [comment]
@mattip Oh wow I completely forgot this thread existed until you commented XD. I think I gave opencv up ages ago. opencv seemed like a nice thing to learn when I first discovered it, so I tried to download it. And then this happened ;-; . I tried to download opencv on my windows pc afther that but for some reason I stopped. I think that was the time I found out about game engines and started learning game programming. Oh well. Building opencv on a pi is indeed really painful like you said :) .
